app/jobs/reservation_job.rb

```diff
@@ -0,0 +1,49 @@
+class ReservationJob < ActiveJob::Base
+  queue_as :default
+  UNDEFINED_MESSAGE = 'JOB NOT DEFINED'.freeze
+
+  def perform
+    if enabled
+      log_start
+      run
+      clean
+      log_completion
+    else
+      log_disabled
+    end
+  end
+
+  private
+
+  def enabled
+    true
+  end
+
+  def log_start(type: '', task: UNDEFINED_MESSAGE)
+    prep_collection
+    Rails.logger.info "Found #{collection.count} #{type} reservations, #{task}"
+  end
```

Comment:
in the case where the task isn't properly setup, maybe `type` should default to the class name? Will need to test it and ensure that it's the child class name (I think it would be but this is the first time I've really used inheritance in ruby in a long while), but I think that it would improve the usefulness of logging when improperly configured